bazel/cc_wrapper.py

```diff
@@ -94,5 +96,72 @@ def main():
   os.execv(compiler, [compiler] + argv)
 
 
+def new_modify_driver_command_line_and_execute(flagfile_path, new_flagfile_path, new_flagfile_fd):
+  flagfile_contents = open(flagfile_path, "r").readlines()
+
+  # Detect if we're building for C++ or vanilla C.
+  if any(map(new_is_cpp_flag, flagfile_contents)):
+    compiler = envoy_real_cxx
+    # Append CXXFLAGS to all C++ targets (this is mostly for dependencies).
+    argv = shlex.split(envoy_cxxflags)
+  else:
+    compiler = envoy_real_cc
+    # Append CFLAGS to all C targets (this is mostly for dependencies).
+    argv = shlex.split(envoy_cflags)
+
+  # Either:
+  # a) remove all occurrences of -lstdc++ (when statically linking against libstdc++),
+  # b) replace all occurrences of -lstdc++ with -lc++ (when linking against libc++).
+  if "-static-libstdc++\n" in flagfile_contents or "-stdlib=libc++" in envoy_cxxflags:
+    for arg in flagfile_contents:
+      if arg == "-lstdc++\n":
+        if "-stdlib=libc++" in envoy_cxxflags:
+          argv.append("-lc++\n")
+      else:
+        argv.append(arg)
+  else:
+    argv += flagfile_contents
+
+  # Bazel will add -fuse-ld=gold in some cases, gcc/clang will take the last -fuse-ld argument,
+  # so whenever we see lld once, add it to the end.
+  if "-fuse-ld=lld\n" in argv:
+    argv.append("-fuse-ld=lld\n")
+
+  # Add compiler-specific options
+  if "clang" in compiler:
+    # This ensures that STL symbols are included.
+    # See https://github.com/envoyproxy/envoy/issues/1341
+    argv.append("-fno-limit-debug-info\n")
+    argv.append("-Wthread-safety\n")
+    argv.append("-Wgnu-conditional-omitted-operand\n")
+  elif "gcc" in compiler or "g++" in compiler:
+    # -Wmaybe-initialized is warning about many uses of absl::optional. Disable
+    # to prevent build breakage. This option does not exist in clang, so setting
+    # it in clang builds causes a build error because of unknown command line
+    # flag.
+    # See https://github.com/envoyproxy/envoy/issues/2987
+    argv.append("-Wno-maybe-uninitialized\n")
+
+  for arg in argv:
+    os.write(new_flagfile_fd, arg)
+  os.execv(compiler, [compiler] + ["@" + new_flagfile_path])
+
+
+def main():
+  # Append CXXFLAGS to correctly detect include paths for either libstdc++ or libc++.
+  if sys.argv[1:5] == ["-E", "-xc++", "-", "-v"]:
+    os.execv(envoy_real_cxx, [envoy_real_cxx] + sys.argv[1:] + shlex.split(envoy_cxxflags))
+
+  if sys.argv[1].startswith("@"):
+    (new_flagfile_fd, new_flagfile_path) = tempfile.mkstemp(dir="./", suffix=".linker-params")
+    flagfile_path = sys.argv[1][1:]
+    with closing_fd(new_flagfile_fd):
+      new_modify_driver_command_line_and_execute(flagfile_path, new_flagfile_path, new_flagfile_fd)
+  else:
+    # TODO(https://github.com/bazelbuild/bazel/issues/7687): Remove this branch
+    # when Bazel 0.27 is released.
+    old_modify_driver_command_line_and_execute(sys.argv)
```

Comment:
Prefer naming `modify_driver..` and `deprecated_modify_driver..`

Reply:
Done.